src/strategies/bearerTokenStrategy.ts

```diff
@@ -2,35 +2,69 @@
  * Builds and configure a Passport strategy to authenticate the proxy clients.
  */
 
+import * as express from "express";
 import { Either } from "fp-ts/lib/Either";
 import * as passport from "passport-http-bearer";
+import { IVerifyOptions } from "passport-http-bearer";
 import container, { SESSION_STORAGE } from "../container";
 import { ISessionState, ISessionStorage } from "../services/ISessionStorage";
-import { SessionToken } from "../types/token";
+import { SessionToken, WalletToken } from "../types/token";
 
 const bearerTokenStrategy = () => {
-  return new passport.Strategy((token, done) => {
+  const options = {
+    passReqToCallback: true,
+    realm: "Proxy API",
+    scope: "request"
+  };
+  return new passport.Strategy(options, (
+    req: express.Request,
+    token: string,
```

Comment:
`string` -> `SessionToken | WalletToken` ?

Reply:
passport.Strategy is expecting a string for the verify callback

```
interface VerifyFunctionWithRequest {
    (req: express.Request, token: string, done: (error: any, user?: any, options?: IVerifyOptions | string) => void): void;
}
```